commitizen/cli.py

```diff
@@ -419,6 +420,23 @@ def main():
             raise NoCommandFoundError()
         raise e
 
+    arguments = vars(args)
+    if unknown_args:
+        # Raise error for extra-args without -- separation
+        if "--" not in unknown_args:
+            raise InvalidCommandArgumentError(
+                f"Invalid commitizen arguments were found: `{' '.join(unknown_args)}`. "
+                "Please use -- separator for extra git args"
+            )
+        # Raise error for extra-args before --
+        elif unknown_args[0] != "--":
+            pos = unknown_args.index("--")
+            raise InvalidCommandArgumentError(
+                f"Invalid commitizen arguments were found before -- separator: `{' '.join(unknown_args[:pos])}`. "
+            )
+        extra_args = " ".join(unknown_args[1:])
```

Comment:
We might need to handle the case that the user use `--` without actually providing an argument. e.g., `cz commit --`

Reply:
Sure, how should we deal with it?

Warning or error?

Reply:
Warning should be enough in this case 👍